multibranch/src/main/java/org/jenkinsci/plugins/workflow/multibranch/SCMVar.java

```diff
@@ -62,6 +65,13 @@
         Job<?,?> job = build.getParent();
         BranchJobProperty property = job.getProperty(BranchJobProperty.class);
         if (property == null) {
+            if (job instanceof WorkflowJob) {
+                FlowDefinition defn = ((WorkflowJob) job).getDefinition();
+                if (defn instanceof CpsScmFlowDefinition) {
+                    // JENKINS-31386: retrofit to work with standalone projects, minus the exact revision support.
```

Comment:
See the series of tests in `SCMBinderTest`. From a multibranch project, `scm` is customized for this build to refer to the exact revision from which `Jenkinsfile` itself was loaded, allowing you to make atomic commits with confidence.

From `CpsScmFlowDefinition` that is not possible (`SCM` provides no suitable API—only multibranch-aware `SCMSource`), so `scm` will be the same configuration as is statically in the project definition, meaning the checkout could be of a newer revision in the same branch (or whatever).